File: base/pythonnet/splitting-excel-files-into-multiple-files/1.py
import aspose.cells as ac
import os

# Define the data directory and file paths
dataDir = "data/"
sourcePath = os.path.join(dataDir, "book1.xls")
outputPath = os.path.join(dataDir, "outputrange.xls")

# Open the source Excel file
sourceWorkbook = ac.Workbook(sourcePath)

# Get the first worksheet from the source workbook
sourceWorksheet = sourceWorkbook.worksheets[0]

# Define the source cell range A1:C10 (10 rows, 3 columns starting at row 0, col 0)
sourceRange = sourceWorksheet.cells.create_range(0, 0, 10, 3)

# Create a new destination workbook
destWorkbook = ac.Workbook()

# Access the first worksheet in the destination workbook
destWorksheet = destWorkbook.worksheets[0]

# Create the destination range at A1 with the same dimensions as the source range
destRange = destWorksheet.cells.create_range(0, 0, 10, 3)

# Copy the source range to the destination range
destRange.copy(sourceRange)

# Save the destination workbook to a new .xls file
destWorkbook.save(outputPath, ac.SaveFormat.EXCEL97_TO2003)

File: base/net/splitting-excel-files-into-multiple-files/1.cs
using System;
using System.IO;
using Aspose.Cells;

// Define the data directory and file paths
string dataDir = "data/";
string sourcePath = dataDir + "book1.xls";
string outputPath = dataDir + "outputrange.xls";

// Open the source Excel file
Workbook sourceWorkbook = new Workbook(sourcePath);

// Get the first worksheet from the source workbook
Worksheet sourceWorksheet = sourceWorkbook.Worksheets[0];

// Define the source cell range A1:C10 (10 rows, 3 columns starting at row 0, col 0)
var sourceRange = sourceWorksheet.Cells.CreateRange(0, 0, 10, 3);

// Create a new destination workbook
Workbook destWorkbook = new Workbook();

// Access the first worksheet in the destination workbook
Worksheet destWorksheet = destWorkbook.Worksheets[0];

// Create the destination range at A1 with the same dimensions as the source range
var destRange = destWorksheet.Cells.CreateRange(0, 0, 10, 3);

// Copy the source range to the destination range
destRange.Copy(sourceRange);

// Save the destination workbook to a new .xls file
destWorkbook.Save(outputPath, SaveFormat.Excel97To2003);